admin_workspace.go

```diff
@@ -0,0 +1,114 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ AdminWorkspaces = (*adminWorkspaces)(nil)
+
+// AdminWorkspaces describes all the admin workspace related methods that the Terraform Enterprise API supports.
+// Note that admin settings are only available in Terraform Enterprise.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/admin/workspaces.html
+type AdminWorkspaces interface {
+	// List all the workspaces within a workspace.
+	List(ctx context.Context, options AdminWorkspaceListOptions) (*AdminWorkspaceList, error)
+
+	// Read a workspace by its ID.
+	Read(ctx context.Context, workspaceID string) (*AdminWorkspace, error)
+
+	// Delete a workspace by its ID.
+	Delete(ctx context.Context, workspaceID string) error
+}
+
+// adminWorkspaces implements AdminWorkspaces.
+type adminWorkspaces struct {
+	client *Client
+}
+
+// AdminWorkspaces represents a Terraform Enterprise admin workspace.
+type AdminWorkspace struct {
+	ID      string   `jsonapi:"primary,workspaces"`
+	Name    string   `jsonapi:"attr,name"`
+	Locked  bool     `jsonapi:"attr,locked"`
+	VCSRepo *VCSRepo `jsonapi:"attr,vcs-repo"`
```

Comment:
It looks like the API only returns the `identifier` for the this and not all the fields of this type will be populated; should we use some other simple type besides `*VCSRepo` to denote this, or would we be boxing ourselves in a corner doing something like `map[string]interface{}`? Whatcha think?

Reply:
Good catch! Looking at the serializers in TFC, `vcs_repo` is only used once for all Admin endpoints, so I think just nesting the struct here would do. 

```
// AdminWorkspaces represents a Terraform Enterprise admin workspace.
type AdminWorkspace struct {
	ID      string `jsonapi:"primary,workspaces"`
	Name    string `jsonapi:"attr,name"`
	Locked  bool   `jsonapi:"attr,locked"`
	VCSRepo *struct {
		Identifier string `json:"identifier"`
	} `jsonapi:"attr,vcs-repo"`
}
```